### PR TITLE
Update README.md

### LINKED ISSUE #789: README issues
- https://github.com/unicode-org/message-format-wg?tab=readme-ov-file#messageformat-2-specification-and-syntax "specification is here (link to follow)" - link is broken
- https://github.com/unicode-org/message-format-wg?tab=readme-ov-file#implementations implementations - mentions ICU 72 (2022) and talks about ICU 75 in the future tense whereas ICU 75 was released last month.

### PR DESCRIPTION
**Pull request by aphillips**
Fixes #789

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ user interfaces that can appear in any language and support the needs of diverse
 
 The current specification starts [here](spec/README.md) and may have changed since the publication
 of the Tech Preview version.
-The Tech Preview specification is [here](tr35-messageformat.md) (link to follow).
+The Tech Preview specification is [here](https://www.unicode.org/reports/tr35/tr35-messageFormat.html)
```
Comment (srl295) [inline README.md:57]: This link will go to whatever the current released version of LDML is. Great if that's what's intended.

If you want the exact version released with CLDR 45 it is here: <https://www.unicode.org/reports/tr35/tr35-72/tr35-messageFormat.html>

File: README.md [hunk 2]
```diff
@@ -107,11 +107,13 @@ Implementers should be aware of the following normative changes during the tech
...
+- Java: [`com.ibm.icu.message2`](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4j/index.html?com/ibm/icu/message2/package-summary.html), part of ICU 75, is a _tech preview_ implementation of the MessageFormat 2 syntax, together with a formatting API. See the [ICU User Guide](https://unicode-org.github.io/icu/userguide/format_parse/messages/mf2.html) for examples and a quickstart guide.
+- C/C++: [`icu::message2::MessageFormatter`](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1message2_1_1MessageFormatter.html), part of ICU 75, is a _tech preview_ implementation of MessageFormat 2.
 - JavaScript: [`messageformat`](https://github.com/messageformat/messageformat/tree/master/packages/mf2-messageformat) 4.0 implements the MessageFormat 2 syntax, together with a polyfill of the runtime API proposed for ECMA-402.
 
+The working group is also aware of these implementations in progress or released, but has not evaluated them:
+- [i18next](https://www.npmjs.com/package/i18next-mf2) i18nFormat plugin to use mf2 format with i18next, version 0.1.1
+
```
Comment (srl295) [inline README.md:116]: Would it be worth adding a call to PR with other implementations as they are known?

**Review by srl295 — APPROVED** [review]
LGTM.  Also, if the CLDR homepage here https://cldr.unicode.org/#h.ezpykkomyltl should link here, let me know.

File: README.md [hunk 3]
```diff
@@ -107,11 +107,17 @@ Implementers should be aware of the following normative changes during the tech
...
+> [!NOTE]
+> Tell us about your MessageFormat 2 implementation!
+> Submit a PR on this page, file an issue, or send email to have your implementation appear here.
```
Comment (srl295) [inline README.md:119]:
```suggestion
> Submit a [PR on this page](https://github.com/unicode-org/message-format-wg/edit/main/README.md), file an issue, or send email to have your implementation appear here.
```

**Review by srl295 — APPROVED** [review]

**Review by eemeli — APPROVED** [review]

**Pull request merged**